### DISCUSSION TITLE
How to use a specific OS image in github actions?

### DISCUSSION THREAD

**Discussion by Zibri** [actions/virtual-environments#8802] [General questions]
Lately the "-latest" image for windows broke a number of scripts I am using.
I wish to use the image as it was a month ago or earlier.
How?

**Comment by Alexey-Ayupov** [comment]
Hello @Zibri, unfortunately, it is not possible. Please refer to this discussion https://github.com/actions/runner-images/discussions/7521

**Comment by Zibri** [comment]
that's too bad. I thionk there should be a mechanism to specify the version. It often happens that an update breaks compilation scripts or worse.
Alternatively you could create "immutable" image along with the updated one. In this way once scripts work with the immutable version they will always work.

**Reply by vpolikarpov-akvelon** [reply]
Hey @Zibri, it's not that simple, unfortunately. Firstly, there are security fixes that we can't ignore. Secondly, a frozen image would become outdated very quickly, diminishing its worthiness. We're doing our best to maintain both stability and updates simultaneously, but sometimes even pigs can fly. If you strictly need to pin the version of any software in your workflow, I would suggest investigating the capabilities of self-hosted runners.
